web/ui/react-app/src/pages/Services.tsx

```diff
@@ -2,14 +2,113 @@ import React, { FC } from 'react';
 import { RouteComponentProps } from '@reach/router';
 import PathPrefixProps from '../PathPrefixProps';
 import { Alert } from 'reactstrap';
+import { useFetch } from '../utils/useFetch';
+import { LabelsTable } from './LabelsTable';
 
-const Services: FC<RouteComponentProps & PathPrefixProps> = ({ pathPrefix }) => (
-  <>
-    <h2>Service Discovery</h2>
-    <Alert color="warning">
-      This page is still under construction. Please try it in the <a href={`${pathPrefix}/service-discovery`}>Classic UI</a>.
-    </Alert>
-  </>
-);
+interface DiscoveredLabels {
+  address: string;
+  metrics_path: string;
+  scheme: string;
+  job: string;
+  my: string;
+  your: string;
+}
+
+interface ActiveTargets {
+  discoveredLabels: DiscoveredLabels[];
+  labels: any;
+  scrapePool: string;
+  scrapeUrl: string;
+  lastError: string;
+  lastScrape: string;
+  lastScrapeDuration: number;
+  health: string;
+}
+
+interface ServiceMap {
```

Comment:
This one duplicates https://github.com/prometheus/prometheus/blob/213a8fe89a7308e73f22888a963cbf9375217cd6/web/ui/react-app/src/pages/targets/ScrapePoolList.tsx#L11-L14.

We should probably find a good common place for these type definitions, but I guess for now it's ok to duplicate here since it's small. But maybe add a `// TODO: Deduplicate with https://github.com/prometheus/prometheus/blob/213a8fe89a7308e73f22888a963cbf9375217cd6/web/ui/react-app/src/pages/targets/ScrapePoolList.tsx#L11-L14`.

Reply:
sure.

Reply:
Any duplicated interface can be added and consumed from https://github.com/prometheus/prometheus/blob/master/web/ui/react-app/src/types/types.ts